plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineExecutionService.java

```diff
@@ -40,13 +43,25 @@ public PipelineExecutionService(PipelineStore store, ThreadPool threadPool) {
         this.threadPool = threadPool;
     }
 
-    public void execute(Data data, String pipelineId, Listener listener) {
+    public Pipeline getPipeline(String pipelineId) {
         Pipeline pipeline = store.get(pipelineId);
+
         if (pipeline == null) {
-            listener.failed(new IllegalArgumentException(LoggerMessageFormat.format("pipeline with id [{}] does not exist", pipelineId)));
-            return;
+            throw new IllegalArgumentException(LoggerMessageFormat.format("pipeline with id [{}] does not exist", pipelineId));
         }
 
+        return pipeline;
+    }
+
+    public void execute(Data data, String pipelineId, Listener listener) {
```

Comment:
seems like only one of the two execute methods are used from the outside world,  I wonder if these changes were precious attempts of reusing code and maybe are not needed anymore. Or maybe I am missing something :)